File: app/src/main/java/com/example/dagger2demo/dagger/cars/Rim.java
package com.example.dagger2demo.dagger.cars;

import android.util.Log;

public class Rim {
    /*we dont own this class so we cannot annotate with @inject*/
    private static final String TAG = "Rim";
public void printRim(){
    Log.d(TAG, "printRim: Rim is running");
}
}

File: app/src/main/java/com/example/dagger2demo/dagger/cars/Rim.kt
package com.example.dagger2demo.dagger.cars

import android.util.Log
import com.example.dagger2demo.dagger.cars.Rim

class Rim {
    fun printRim() {
        Log.d(TAG, "printRim: Rim is running")
    }

    companion object {
        /*we dont own this class so we cannot annotate with @inject*/
        private const val TAG = "Rim"
    }
}